### PR TITLE
Add basic oAuth2 support

### PR DESCRIPTION
**Pull request by duronrulez**
Allow for basic oAuth2 operations to be performed against the blizzard Community and Game Data endpoints.

### CODE REVIEW

**Comment by LogansUA** [comment]
Sweet, thanks.

File: src/BlizzardClient.php [hunk 1]
```diff
@@ -213,12 +278,33 @@ private function configureOptions(OptionsResolver $resolver, $region)
...
+                 ->setAllowedTypes('apiSecret', 'string')
                  ->setAllowedTypes('region', 'string')
                  ->setAllowedValues('region', array_keys(GeoData::$list))
                  ->setAllowedTypes('locale', 'string')
-                 ->setAllowedValues('locale', $locales)
-                 ->setAllowedTypes('accessToken', ['null', 'string']);
+                 ->setAllowedValues('locale', $locales);
+    }
+
+    protected function requestAccessToken() {
```
Comment (LogansUA) [inline src/BlizzardClient.php]: Reformat function according to PSR standarts, and add DocBlock

File: src/Service/Service.php [hunk 2]
```diff
@@ -53,6 +53,7 @@ protected function request($urlSuffix, array $options)
 
         $result = $client->get($this->serviceParam.$urlSuffix, $options);
 
+        var_dump($result->getBody()->getContents());
```
Comment (LogansUA) [inline src/Service/Service.php]: Remove

File: src/Tokens/Access.php [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+     * Create token from json object
+     * @param \stdClass $jsonObject
+     * @return \BlizzardApi\Tokens\Access
+     */
+    public static function fromJson($jsonObject)
+    {
+        return new self($jsonObject->access_token, $jsonObject->token_type, $jsonObject->expires_in);
+    }
+
+    public function isExpired() {
```
Comment (LogansUA) [inline src/Tokens/Access.php]: Reformat, add DocBlock

File: src/Tokens/Access.php [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        if ($this->expiresAt > new \DateTime()) {
+            return false;
+        }
+
+        return true;
+    }
+
+    public function getToken() {
```
Comment (LogansUA) [inline src/Tokens/Access.php]: Reformat, add DocBlock

**Comment by duronrulez** [comment]
Fixed CS and added DocBlocks

**Comment by LogansUA** [comment]
@duronrulez Cool, thank you. Will fix it and create new Release.

**Pull request merged**